test/test_library_loading.cpp

```diff
@@ -0,0 +1,171 @@
+#include "dlibxx.hxx"
+
+#include <gmock/gmock.h>
+#include "mocks/dlfcn_mock.h"
+
+#include <tuple>
+
+using namespace testing;
+
+class TestLoadLibrary : public Test
+{
+public:
+	TestLoadLibrary()
+		: Test{}
+	{
+		ON_CALL(dlfcn, dlopen(_,_)).WillByDefault(Return(reinterpret_cast<void*>(1)));
+		ON_CALL(dlfcn, dlclose(_)).WillByDefault(Return(0));
+	}
+
+protected:
+	NiceMock<DlfcnMock> dlfcn;
+	dlibxx::handle lib;
+};
+
+TEST_F(TestLoadLibrary, Loaded_WhenLibraryCannotBeFound_WillReturnFalse)
+{
+	auto const lib_name = std::string{ "test"};
```

Comment:
`std::string lib_name = "test";` and for others below.

Reply:
Yes, I can do that. I usually use type on the right since the alignment of variable declarations, as a consequence, becomes more pleasant to look at. Other reasons are basically covered by [Herb Sutter](https://herbsutter.com/2013/08/12/gotw-94-solution-aaa-style-almost-always-auto/). However, I am happy to commit to changing at your request, as well.

Reply:
I don't strongly disagree with any of his arguments, although I'm not a fan of the type being implicit (which isn't the case here), so I think we can leave these as is. Although, any reason for the space between the `{` and `"test"`?

Reply:
No, that space is just preference; though I usually have it on both sides. Slipped through because of my over-reliance on clang-format, that is, for the time being at least, missing here :)

Reply:
I think I mentioned in the review level comment that I was planning on running clang-format across the whole codebase so if you run it over here I'll be happy with whatever it spits out :)

Reply:
Yeah, with Google style. I will run that over the PR.